str/catalog_design/polymorphic_site_extractor.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+# pylint: disable=missing-function-docstring,no-member
+"""
+This script:
+(1) parses a given VCF for polymorphic sites
+(2) filters a given catalog JSON file to only include those sites, outputting the filtered catalog to a new JSON file.
+(3) shards the filtered catalog into chunks, writing out each chunk as a separate JSON file.
+
+Polymorphic sites are those that have at least two distinct alleles occuring in the VCF.
+Note this script filters out chrY and chrM sites by default.
+
+ analysis-runner --dataset "tob-wgs" \
+    --description "polymorphic-site-extractor" \
+    --access-level "test" \
+    --output-dir "hoptan-str/catalog-design/" \
+    --memory=32G --storage=20G \
+    polymorphic_site_extractor.py --vcf-path=gs://cpg-tob-wgs-test/hoptan-str/shard_workflow_test/merge_str_vcf_combiner/combined_eh.vcf \
+    --catalog-path=gs://cpg-tob-wgs-test/hoptan-str/5M_run/5M_sharded_100k/chunk_1.json
+    --folder_name=sharded_polymorphic_catalog
+
+"""
+import json
+import hail as hl
+import click
+
+
+from cpg_utils import to_path
+from cpg_utils.hail_batch import output_path, init_batch
+
+
+def polymorphic_site_extractor(file_path):
+    """Extracts polymorphic sites from a VCF file and returns a list of REPIDs (similar to rsids) representing those sites
+    The output has been benchmarked with Gymrek's statSTR and agrees with the number of polymorphic sites found
+    """
+    init_batch()
+    # read in VCF into mt format
+    mt = hl.import_vcf(file_path)
+
+    mt = hl.sample_qc(mt)
+    mt = hl.variant_qc(mt)
+
+    # remove chrY and chrM
+    filtered_mt = mt.filter_rows(~hl.str(mt.locus.contig).startswith('chrY'))
+    filtered_mt = filtered_mt.filter_rows(
+        ~hl.str(filtered_mt.locus.contig).startswith('chrM')
+    )
+
+    # remove loci that are monomorphic for the REF allele
+    filtered_mt = filtered_mt.filter_rows(hl.len(filtered_mt.alleles) > 1)
+
+    # at a biallelic locus, remove loci that are monomorphic for 1 ALT allele
+    filtered_mt = filtered_mt.filter_rows(
+        ~(
+            (hl.len(filtered_mt.variant_qc.AC) == 2)
+            & (filtered_mt.variant_qc.AC[0] == 0)
+        )
+    )
+
+    # collect the REPIDs into one list
+    rep_id_list = filtered_mt.info.REPID.collect()
+
+    return rep_id_list
+
+
+def catalog_filter(rep_id_list, catalog_path, gcs_output_path):
+    """Retains loci in a JSON file that intersect with a list of REPIDs (STR equiv. of rsids) and writes the filtered catalog to a new JSON file"""
+    with to_path(catalog_path).open('r') as json_file:
+        # Load the JSON content
+        catalog = json.load(json_file)
+
+    filtered_data = []
+
+    # Make the rep_id_list into a set
+    polymorphic_rep_id_set = set(rep_id_list)
```

Comment:
This is just removing the casting of a list to set, and making a corresponding change to the variable name
```suggestion
def catalog_filter(polymorphic_rep_id_set: set[str], catalog_path: str, gcs_output_path: str):
    """Retains loci in a JSON file that intersect with a list of REPIDs (STR equiv. of rsids) and writes the filtered catalog to a new JSON file"""
    with to_path(catalog_path).open('r') as json_file:
        # Load the JSON content
        catalog = json.load(json_file)

    filtered_data = []
```